daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SExpr.scala

```diff
@@ -93,88 +95,75 @@ object SExpr {
           SPAP(PBuiltin(b), new util.ArrayList(), b.arity)
       }
     }
-    def execute(machine: Machine): Unit = {
-      /* special case for nullary record constructors */
-      machine.returnValue = b match {
-        case SBRecCon(id, fields) if b.arity == 0 =>
-          SRecord(id, fields, new util.ArrayList())
-        case _ =>
-          SPAP(PBuiltin(b), new util.ArrayList(), b.arity)
-      }
-    }
   }
 
   /** A pre-computed value, usually primitive literal, e.g. integer, text, boolean etc. */
   final case class SEValue(v: SValue) extends SExprAtomic {
-    def evaluate(machine: Machine): SValue = {
+    def lookupValue(machine: Machine): SValue = {
       v
     }
-    def execute(machine: Machine): Unit = {
-      machine.returnValue = v
-    }
   }
 
   object SEValue extends SValueContainer[SEValue]
 
-  /** Function application:
-    General case: 'fun' and 'args' are any kind of expression */
+  /** Function application: General case: 'fun' and 'args' are any kind of expression */
   final case class SEAppGeneral(fun: SExpr, args: Array[SExpr]) extends SExpr with SomeArrayEquals {
     def execute(machine: Machine): Unit = {
-      machine.pushKont(KArg(args, machine.frame, machine.actuals, machine.env.size))
-      machine.ctrl = fun
+      //machine.pushKont(KArg(args, machine.frame, machine.actuals, machine.env.size))
+      //machine.ctrl = fun
```

Comment:
What's the value of leaving this around?
```suggestion
```

Reply:
it's gone